actors/builtin/miner/miner_actor.go

```diff
@@ -1610,6 +1611,24 @@ func handleProvingDeadline(rt Runtime) {
 				st.ProvingPeriodStart = st.ProvingPeriodStart + WPoStProvingPeriod
 			}
 		}
+
+		// expire pre-committed sectors
+		expiryQ, err := LoadBitfieldQueue(store, st.PreCommittedSectorsExpiry, quant)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sector expiry queue")
+
+		bf, modified, err := expiryQ.PopUntil(currEpoch)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pop expired sectors")
+
+		if modified {
+			st.PreCommittedSectorsExpiry, err = expiryQ.Root()
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to save expiry queue")
+		}
+
+		if bf != nil {
+			depositToBurn, err := st.checkPrecommitExpiry(store, bf)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to expire pre-committed sectors")
+			burnFunds(rt, depositToBurn)
+		}
```

Comment:
Let's move this whole lot to the top, immediately after vesting funds.

Push it down into a method on miner.State that returns the total penalty.

Reply:
@anorth Not sure what you mean here. Do you mean push the pre commit expiry part in a separate func ?